### PR TITLE
Matter: add reference to Thread docs

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Added a guidance note for users implementing OpenThread as a border router for Matter devices, emphasizing the importance of consulting specific Thread documentation for proper configuration.
	- Updated instructions to include the option of using Home Assistant with the OpenThread Border Router add-on as a valid Thread border router device for both Android and iPhone users.

### CODE REVIEW

**Review by agners — DISMISSED** [review]

File: source/_integrations/matter.markdown [hunk 1]
```diff
@@ -138,10 +138,12 @@ Make sure you have all these components ready before trying to add a Matter devi
     - Have the Android version 8.1 or higher.
     - Have the latest version of the Home Assistant Companion app, installed from the Play Store (full version).
     - If you are using {% term Thread %}: Make sure there is a Thread border router device (Nest Hub (2nd Gen) or Nest Wi-Fi Pro) present in your home network.
+      - If you are using OpenThread (for Connect ZBT-1/SkyConnect) as border router, make sure you followed the steps in the [Thread documentation](https://www.home-assistant.io/integrations/thread#turning-home-assistant-into-a-thread-border-router).
```
Comment (c0ffeeca7) [inline source/_integrations/matter.markdown]:
```suggestion
      - If you are using OpenThread (for Connect ZBT-1/SkyConnect) as border router, make sure you followed the steps in the [Thread documentation](/integrations/thread#turning-home-assistant-into-a-thread-border-router).
```

File: source/_integrations/matter.markdown [hunk 2]
```diff
@@ -138,10 +138,12 @@ Make sure you have all these components ready before trying to add a Matter devi
...
   - iPhone
     - Have the iOS version 16 or higher
     - Have the latest version of the Home Assistant Companion app installed.
     - If you are using {% term Thread %}: Make sure there is a Thread border router device (HomePod Mini or V2, Apple TV 4K) present in your home network.
+      - If you are using OpenThread (for Connect ZBT-1/SkyConnect) as border router, make sure you followed the steps in the [Thread documentation](https://www.home-assistant.io/integrations/thread#turning-home-assistant-into-a-thread-border-router).
```
Comment (c0ffeeca7) [inline source/_integrations/matter.markdown]:
```suggestion
      - If you are using OpenThread (for Connect ZBT-1/SkyConnect) as border router, make sure you followed the steps in the [Thread documentation](/integrations/thread#turning-home-assistant-into-a-thread-border-router).
```

File: source/_integrations/matter.markdown [hunk 3]
```diff
@@ -138,10 +138,12 @@ Make sure you have all these components ready before trying to add a Matter devi
     - Have the Android version 8.1 or higher.
     - Have the latest version of the Home Assistant Companion app, installed from the Play Store (full version).
     - If you are using {% term Thread %}: Make sure there is a Thread border router device (Nest Hub (2nd Gen) or Nest Wi-Fi Pro) present in your home network.
```
Comment (agners) [inline source/_integrations/matter.markdown]:
```suggestion
    - If you are using {% term Thread %}: Make sure there is a Thread border router device (Nest Hub (2nd Gen) or Nest Wi-Fi Pro or Home Assistant with the OpenThread Border Router add-on) present in your home network.
```

File: source/_integrations/matter.markdown [hunk 4]
```diff
@@ -138,10 +138,12 @@ Make sure you have all these components ready before trying to add a Matter devi
...
+      - If you are using OpenThread (for Connect ZBT-1/SkyConnect) as border router, make sure you followed the steps in the [Thread documentation](/integrations/thread#turning-home-assistant-into-a-thread-border-router).
   - iPhone
     - Have the iOS version 16 or higher
     - Have the latest version of the Home Assistant Companion app installed.
     - If you are using {% term Thread %}: Make sure there is a Thread border router device (HomePod Mini or V2, Apple TV 4K) present in your home network.
```
Comment (agners) [inline source/_integrations/matter.markdown]:
```suggestion
    - If you are using {% term Thread %}: Make sure there is a Thread border router device (HomePod Mini or V2, Apple TV 4K or Home Assistant with the OpenThread Border Router add-on) present in your home network.
```

**Review by agners — CHANGES_REQUESTED** [review]
Actually, looking it again it is weird that we have a sub-point which mentions OpenThread, but it is not a option in the parent item. Let's add OTBR to the list there too.

**Review by agners — APPROVED** [review]
LGTM, thanks!

**Pull request merged**